sdk-cpp/build/cmake/deliveryoptimization_sdk-config.cmake

```diff
@@ -1,11 +1,65 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
+macro(override_prefix_path_if_needed)
+    if (CMAKE_SYSTEM_NAME MATCHES Linux)
+        # Ubuntu 20.04 symlinks /lib to /usr/lib but there is no equivalent /include path.
+        # This causes FindBoost to incorrectly set the headers path to the non-existent /include path.
+        # Work around it by temporarily updating CMAKE_PREFIX_PATH.
+        if (EXISTS "/lib" AND NOT EXISTS "/include")
+            if (CMAKE_PREFIX_PATH)
+                message (WARNING "/lib exists but not /include, and CMAKE_PREFIX_PATH is already set. FindBoost may result in non-existent include dir.")
+            else ()
+                message (WARNING "Updating cmake prefix path from '${CMAKE_PREFIX_PATH}' to '/usr' temporarily for FindBoost.")
+                set(DO_CMAKE_PREFIX_PATH_SAVE ${CMAKE_PREFIX_PATH})
+                set(CMAKE_PREFIX_PATH "/usr")
+            endif ()
+        endif ()
+    endif ()
+endmacro()
+
+macro(restore_prefix_path_if_needed)
+    if (DO_CMAKE_PREFIX_PATH_SAVE)
+        message (STATUS "Restoring cmake prefix path to '${DO_CMAKE_PREFIX_PATH_SAVE}'")
+        set(CMAKE_PREFIX_PATH ${DO_CMAKE_PREFIX_PATH_SAVE})
+        # Cleanup temporary variable
+        set(DO_CMAKE_PREFIX_PATH_SAVE)
+    endif ()
+endmacro()
+
+macro(fixup_do_includes_if_needed)
+    if (CMAKE_SYSTEM_NAME MATCHES Linux)
+        # See explanation in macro override_prefix_path_if_needed, above.
+        # Setting CMAKE_PREFIX_PATH does not work for DO SDK though. So update the INTERFACE_INCLUDE_DIRECTORIES property directly.
+        get_target_property(DOSDK_INC_DIR_PRIV Microsoft::deliveryoptimization INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
>DOSDK_INC_DIR_PRIV [](http://example.com/codeflow?start=28&length=18)

where is this set?

Reply:
That is the output variable for get_target_property method call.

Reply:
thx, not super familiar with cmake.